interfaces/policy/basedeclaration.go

```diff
@@ -658,9 +669,48 @@ slots:
         - core
 `
 
+func trimTrailingNewline(s string) string {
+	return strings.TrimRight(s, "\n")
+}
+
+func ComposeBaseDeclaration(ifaces []interfaces.Interface) ([]byte, error) {
```

Comment:
why public and without doc? it could be private and then tested using export_test?